resource_retriever/include/resource_retriever/retriever.hpp

```diff
@@ -65,26 +53,20 @@ struct MemoryResource
 class RESOURCE_RETRIEVER_PUBLIC Retriever
 {
 public:
-  Retriever();
+  explicit Retriever(RetrieverVec plugins = default_plugins());
 
   ~Retriever();
 
-  Retriever(const Retriever & ret) = delete;
-  Retriever & operator=(const Retriever & other) = delete;
-
-  Retriever(Retriever && other) noexcept;
-  Retriever & operator=(Retriever && other) noexcept;
-
   /**
    * \brief Get a file and store it in memory
    * \param url The url to retrieve.  package://package/file will be turned into the correct file:// invocation
    * \return The file, loaded into memory
    * \throws resource_retriever::Exception if anything goes wrong.
    */
-  MemoryResource get(const std::string & url);
+  MemoryResourcePtr get(const std::string & url);
```

Comment:
This is an important change to the API, going from returning a structure to returning a shared_ptr.  Given that we usually tick-tock this stuff, can we add back the old API and mark it as deprecated?